File: src/main/java/com/model/Fund.java
package com.model;

import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.Fetch;
import org.hibernate.annotations.FetchMode;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;
import java.io.Serializable;
import java.util.List;

@Entity
@Table(name = "funds")
@Setter
@Getter
public class Fund implements Serializable {

    @Id
    @GeneratedValue(generator = "uuid")
    @GenericGenerator(name = "uuid", strategy = "uuid")
    private String id;

    @Column
    private String name;

    @Fetch(FetchMode.JOIN)
    @OneToMany(mappedBy = "fundId")
    private List<Item> items;
}

File: ui/src/app/models/fund.ts
import { Item } from './item';

export class Fund {
  id: string;
  name: string;
  amount?: string;
  income?: string;
  expense?: string;
  items: Array<Item>;
  startDate: string;

  constructor( name: string = null,
               items: Array<Item> = null,
               startDate = null,
               id: string = null ) {
    this.id = id;
    this.name = name;
    this.items = items;
    this.startDate = startDate;
  }
}
